Sources/xcproj/PBXProj+Helpers.swift

```diff
@@ -98,6 +98,10 @@ extension PBXProj {
 extension PBXProj: Writable {
     
     public func write(path: Path, override: Bool) throws {
+        if let projectName = path.projectName() {
```

Comment:
I don't think we need this here. If we already set the name when we read the project, that should be enough. Moreover, I don't think this method would work because the path that we are passing here is the `.pbxproj` path, not the project path, so the name that you would get here is `project` which is the default `pbxproj` name.

Reply:
I believe this part is actually more important than reading. When we write project to Path, the name of the project will come form the directory where `pbxproj` file is in, not from the `PBXProj` attribute `name`.

For examlpe. When you load load project from `Project.xcodeproj/project.pbxproj` and write it to `MyProject.xcodeproj/project.pbxproj` then the comments should be for `MyProject`. Otherwise when you resave it with Xcode, you will get an unwanted diff.

Reply:
I agree, then I'd keep it in only one place, the writing. What do you think? If I'm not wrong, I think it should be `path.parent.projectName()`, shouldn't it?

Reply:
I changed getting project name from `Path` to `path.parent.projectName()` but kept it in internal extension, as this allows us to cover it with unit test and this will describe the needs/limitations of it.

In addition to writing the project name, we should read it also. It will come in handy when you want to serialise the project to Data, after reading it from file. In this case the original project name is kept in comments.